Kuroba/app/src/main/java/com/github/adamantcheese/chan/core/site/sites/chan55/Chan55.java

```diff
@@ -0,0 +1,308 @@
+                all.add(Board.fromSiteNameCode(Chan55.this, "Idiomas", "lang"));
+                all.add(Board.fromSiteNameCode(Chan55.this, "DIY, gambiarras e projetos", "macgyver"));
+                all.add(Board.fromSiteNameCode(Chan55.this, "Automóveis", "pfiu"));
+                all.add(Board.fromSiteNameCode(Chan55.this, "Culinária", "coz"));
+                all.add(Board.fromSiteNameCode(Chan55.this, "Universidade Federal do 55chan", "UF55"));
+                all.add(Board.fromSiteNameCode(Chan55.this, "Finanças", "$"));
+
+                all.add(Board.fromSiteNameCode(Chan55.this, "Politicamente Incorreto", "pol"));
+                all.add(Board.fromSiteNameCode(Chan55.this, "Esportes", "esp"));
+                all.add(Board.fromSiteNameCode(Chan55.this, "Viadices", "clô"));
+                all.add(Board.fromSiteNameCode(Chan55.this, "Buseta e Depressão", "escoria"));
+
+                all.add(Board.fromSiteNameCode(Chan55.this, "Pornografia 2D", "34"));
+                all.add(Board.fromSiteNameCode(Chan55.this, "*fapfapfap*", "pr0n"));
+                all.add(Board.fromSiteNameCode(Chan55.this, "Pintos Femininos", "tr"));
+                all.add(Board.fromSiteNameCode(Chan55.this, "Ai, que delícia, cara", "pinto"));
+
+                listener.onBoardsReceived(new Boards(all));
+                boardManager.updateAvailableBoardsForSite(Chan55.this, all);
+            }
+
+            @Override
+            public void post(Reply reply, PostListener postListener) {
+                ReplyResponse replyResponse = new ReplyResponse();
+
+                reply.password = Long.toHexString(secureRandom.nextLong());
+                replyResponse.password = reply.password;
+
+                MultipartHttpCall call = new MultipartHttpCall(site) {
+                    @Override
+                    public void process(Response response, String result) {
+                        handlePost(replyResponse, response, result);
+                    }
+
+                    @Override
+                    public void setup(
+                            Request.Builder requestBuilder,
+                            @Nullable ProgressRequestBody.ProgressRequestListener progressListener
+                    ) {
+                        super.setup(requestBuilder, progressListener);
+
+                        // Change referer
+                        ReplyPresenter repPres = (ReplyPresenter) postListener;
+                        requestBuilder.removeHeader("Referer");
+                        requestBuilder.addHeader("Referer", String.format("%s/%s/res/%s.html", BASE_URL, repPres.getBoard().code, String.valueOf(repPres.getCurrentOpNo())));
+                    }
+                };
+
+                call.url(site.endpoints().reply(reply.loadable));
+
+                if (requirePrepare()) {
+                    Handler handler = new Handler(Looper.getMainLooper());
```

Comment:
This should be made a class variable too. It shouldn't be created every time this method is called.